ivy-test.el

```diff
@@ -26,32 +26,37 @@
 
 ;;; Code:
 
-(defvar require-features nil)
 (defvar ivy-empty "tests/find-file/empty-dir/")
 
-(defadvice require (before ivy-tests-require-hook (feature &rest _) activate)
-  "Record the requires into `require-features'."
-  (push feature require-features))
+(defvar ivy-features nil
+  "Like `features' but for Ivy testing purposes.")
 
-(require 'ert)
-(require 'colir)
+(defun ivy-test--record-feature (feature &rest _)
+  "Record FEATURE in `ivy-features'.
+Intended as :after-while advice for `require'."
+  (add-to-list 'ivy-features feature nil #'eq))
```

Comment:
I worry this might impact the test time, as it is linear in the number of required packages.

Reply:
I'm not worried, and here's why: every time a Lisp library calls `(require 'foo)`, the feature `foo` is looked up in the list `features` using `memq`. If the performance of this is good enough for Emacs in normal use, then it's good enough for Ivy tests.

More concretely, in my interactive Emacs session, `(length features)` is less than 400, which is pretty negligible. If for whatever reason this ever becomes a performance problem, then a list is not the right data structure to begin with.

Having said that, there's no strong incentive to deduplicate entries in `ivy-features`, but I decided to go with it for consistency with the built-in `features`. Feel free to rever to `push`.